src/material/form-field/form-field.html

```diff
@@ -16,12 +16,12 @@
   -->
   <label matFormFieldFloatingLabel
          [floating]="_shouldLabelFloat()"
+         [monitorResize]="appearance === 'outline'"
          *ngIf="_hasFloatingLabel()"
-         (cdkObserveContent)="_refreshOutlineNotchWidth()"
-         [cdkObserveContentDisabled]="!_hasOutline()"
          [id]="_labelId"
          [attr.for]="_control.id"
-         [attr.aria-owns]="_control.id">
+         [attr.aria-owns]="_control.id"
+         (resized)="_refreshOutlineNotchWidth()">
```

Comment:
I think that this will lead to change detections even though we're running the resize observer outside the `NgZone`. Maybe we can have the notched outline set its own width and avoid having to emit an event in the first place?

Reply:
I just changed it to not go through a template event binding. I think that should do it

Reply:
On second thought, it looks like that's not so simple because the components class gets the floating label and outline with `@ViewChild`, but they actually change if appearance is toggled. Would probably need to change them to `@ViewChildren` and watch for the latest one, but detangling that is not a rabbit hole I want to go down in this PR.

Reply:
Instead of getting the floating label through `ViewChild`, we can have the floating label inject the form field and notify it when a resize has happened.